evalml/pipelines/component_graph.py

```diff
@@ -146,6 +146,8 @@ def _fit_transform_features_helper(self, needs_fitting, X, y=None):
                 parent_output = infer_feature_types(parent_output)
             final_component_inputs.append(parent_output)
         concatted = pd.concat([component_input.to_dataframe() for component_input in final_component_inputs], axis=1)
+        if needs_fitting:
```

Comment:
since we call `self._compute_features(self.compute_order[:-1]...`, we miss out on the last element (usually estimator) in `self.input_feature_names`. Manually adding here.

Reply:
Can you help me understand how this ties into the issue with the metaclasses?

Reply:
It doesn't--it was a bug I noticed while writing the tests and trying to call `feature_importance` on time series pipelines 😅